packages/core/src/web/helpers/fonts/googleFontService.ts

```diff
@@ -0,0 +1,259 @@
+
+  if (directMatch) {
+    return true;
+  }
+
+  return false;
+};
+
+/**
+ * Clean Google Fonts from font history when offline to prevent future loading attempts
+ * Keeps only local and web-safe fonts in the history
+ */
+const cleanGoogleFontsFromHistory = (fontHistory: string[]): void => {
+  const fontStore = useGoogleFontStore.getState();
+  const cleanedHistory = fontHistory.filter((family) => isLocalFont(family) || fontStore.isWebSafeFont(family));
+
+  if (cleanedHistory.length !== fontHistory.length) {
+    useStorageStore.getState().set('font-history', cleanedHistory);
+  }
+};
+
+/**
+ * Load Google Fonts from font history when online, skip and clean history when offline
+ */
+const loadHistoryGoogleFonts = (): void => {
+  const fontHistory = useStorageStore.getState()['font-history'];
+
+  if (!fontHistory || !Array.isArray(fontHistory) || fontHistory.length === 0) {
+    return;
+  }
+
+  const fontStore = useGoogleFontStore.getState();
+  const networkAvailable = fontStore.isNetworkAvailableForGoogleFonts();
+
+  if (!networkAvailable) {
+    cleanGoogleFontsFromHistory(fontHistory);
+
+    return;
+  }
+
+  const googleFontsFromHistory = fontHistory.filter((family) => !isLocalFont(family));
+
+  if (googleFontsFromHistory.length === 0) {
+    return;
+  }
+
+  googleFontsFromHistory.forEach((family) => {
+    fontStore.loadGoogleFont(family);
+  });
+};
+
+/**
+ * Replace Google Fonts with web-safe fallbacks in SVG text elements
+ */
+const applyGoogleFontFallbacks = (textElements: SVGTextElement[], store: any): void => {
+  let fallbackCount = 0;
+  const replacements: Array<{ from: string; to: string }> = [];
+
+  textElements.forEach((textElem) => {
+    const fontFamily = textElem.getAttribute('font-family');
+
+    if (fontFamily) {
+      const cleanFamily = fontFamily.replace(/^['"]+|['"]+$/g, '').trim();
+
+      if (store.isWebSafeFont(cleanFamily) || isLocalFont(cleanFamily)) {
+        return;
+      }
+
+      const fallbackFont = store.getFallbackFont(cleanFamily);
+      const fallbackPostScriptName = store.getFallbackPostScriptName(fallbackFont);
+
+      textElem.setAttribute('font-family', fallbackFont);
+
+      const postScriptName = textElem.getAttribute('font-postscript-name');
+
+      if (postScriptName) {
+        textElem.setAttribute('font-postscript-name', fallbackPostScriptName);
+      }
+
+      const dataPostScriptName = textElem.getAttribute('data-postscript-name');
+
+      if (dataPostScriptName) {
+        textElem.setAttribute('data-postscript-name', fallbackPostScriptName);
+      }
+
+      const allAttributes = textElem.attributes;
+
+      for (const attr of allAttributes) {
+        if (
+          attr.name.toLowerCase().includes('postscript') &&
+          attr.name !== 'font-postscript-name' &&
+          attr.name !== 'data-postscript-name'
+        ) {
+          textElem.setAttribute(attr.name, fallbackPostScriptName);
+        }
+      }
```

Comment:
Is there any reason to handle `'font-postscript-name'` and `'data-postscript-name'` separately?
```
for (const attr of allAttributes) {
    if (attr.name.toLowerCase().includes('postscript')) {
        textElem.setAttribute(attr.name, fallbackPostScriptName);
    }
}
```
Look enough to me